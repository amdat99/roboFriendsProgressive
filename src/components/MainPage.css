/* #### Generated By: http://www.cufonfonts.com #### */

@font-face {
	font-display: swap;
font-family: 'SEGA LOGO FONT';
font-style: normal;
font-weight: normal;
src: local('SEGA LOGO FONT'), url('SEGA.woff') format('woff');
}

h1 {
  font-family: 'SEGA LOGO FONT';
  font-weight: 200;
  color: #16DA5A;
  text-shadow: 2px 2px #ff0000;

;
}