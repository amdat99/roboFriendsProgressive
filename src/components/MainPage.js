
import React, { Component } from 'react'; // classes react import
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';
import Header from './Header';

import './MainPage.css';

class MainPage extends Component{
 
   componentDidMount() {

   console.log(process.env.REACT_APP_SAY_HI) // redux class function runs when component loads 
    this.props.onRequestRobots()
  }

   // useEffect(()=>{
   //    console.log(count)
   //     },[count]) // appying count state in array ensures funtion runs when state changes

// const onSearchChange = (event) => { // search change funtion with hooks redux
//   dispatch(setSearchField( event.target.value ))
//   }


  // const onSearchChange = (event) => { // hooks onsearhhangefunction 
  // setSearchfield( event.target.value )
  // }
  filteredRobots = () =>{ //filtered robots function 
  return this.props.robots.filter(robot =>{
    return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    })
 }

  render(){
 const {searchField, onSearchChange, robots, isPending} = this.props; // assigning with redux classes

// const filteredRobots = robots.filter(robot =>{
//     return robot.name.toLowerCase().includes(searchField.toLowerCase());
//     })
      

   return (
      <div className='tc'>
        <Header />
        <SearchBox searchChange={onSearchChange}/>
        <Scroll>
          { isPending ? <h1>Loading</h1> :
            <ErrorBoundry>
            
              <CardList robots={this.filteredRobots()} />
            </ErrorBoundry>
          }
        </Scroll>
      </div>
    );
  } }


export default MainPage;
// export default connect(mapStateToProps, mapDispatchToProps)(App); // how to export using redux classed and how to 