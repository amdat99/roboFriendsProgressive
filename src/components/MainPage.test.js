import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage'

import {createSerializer} from 'enzyme-to-json';
import toJson from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

let wrapper;
beforeEach(() => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false

	}
	wrapper = shallow(<MainPage {...mockProps}/>)
})
it('renders mainpage without crashing', () => {
		const mockColor = 'red'
	// Method1
	expect(toJson(wrapper)).toMatchSnapshot();
	})

	it('filters robots correcty', () =>{
		expect(wrapper.instance().filteredRobots()).toEqual([])
	})


let wrapper2;
beforeEach(() => {
	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id:3,
			name: 'john',
			email: 'john@gmail.com'
		}],
		searchField: 'john',
		isPending: false}
	wrapper2 = shallow(<MainPage {...mockProps2}/>)
})
it('filters robots correcty', () =>{
		expect(wrapper2.instance().filteredRobots()).toEqual([{
			id:3,
			name: 'john',
			email: 'john@gmail.com'}])
	})

let wrapper3;
beforeEach(() => {
	const mockProps3 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id:3,
			name: 'john',
			email: 'john@gmail.com'
		}],
		searchField: 'a',
		isPending: false

	}
	wrapper3 = shallow(<MainPage {...mockProps3}/>)
	
})
		const filteredRobots = []
		it('filters robots correcty2', () =>{
		expect(wrapper3.instance().filteredRobots()).toEqual(filteredRobots)
})

let wrapper4;
beforeEach(() => {
	const mockProps4 = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: true

	}
	wrapper4 = shallow(<MainPage {...mockProps4}/>)
})


	it('recieves loading html if is pending true', () =>{
		expect(wrapper4).toMatchSnapshot();
   })



