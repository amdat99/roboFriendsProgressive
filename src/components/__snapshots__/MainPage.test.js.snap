// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recieves loading html if is pending true 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <h1>
      Loading
    </h1>
  </Scroll>
</div>
`;

exports[`renders mainpage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
